Initial add of Mixerless Telemetry Filter tests

spongeman66 opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | Adds tests for Mixerless Telemetry Filters
| License         | Apache 2.0


### What's in this PR?
Adds tests for Mixerless Telemetry Filters


### Why?
Adds system level tests


### Additional context
This PR is not fully working yet. 

### Checklist
<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->
- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### To Do
<!-- (Please remove this section if you don't need it.) -->
- [x] If the PR is not complete but you want to discuss the approach, list what remains to be done here

Commit:
Initial add of Mixerless Telemetry Filter tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/banzaicloud/istio-operator?pullRequest=662) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/banzaicloud/istio-operator?pullRequest=662) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: khernyo<br/>:white_check_mark: spongeman66<br/>:x: Rod Slattery<br/><hr/>**Rod Slattery** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/banzaicloud/istio-operator?pullRequest=662) it.</sub>

spongeman66 committed:
Initial add of Mixerless Telemetry Filter tests

spongeman66 committed:
Merge branch 'mixerless_1' of github.com:spongeman66/istio-operator into mixerless_1

sagikazarmark commented on test/e2e/e2e_helpers_test.go:
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
+		return false, nil
+	})
+}
+
+//
+//
+//
+//
+//
+//	timer := time.After(timeout)
+//	ticker := time.Tick(interval)
+//	// Keep trying until we're timed out or got a result
+//	log.Info("Waiting for:", "filter", filtername, "expectMissing", expectMissing)
+//	for {
+//		select {
+//		// Got a timeout! fail with a timeout error
+//		case <-timer:
+//			return false, errors.NewWithDetails("Timeout waiting for expected filter state")
+//		// Got a tick, we should check on doSomething()
+//		case <-ticker:
+//			_, err := testEnv.Dynamic.Resource(gvr.EnvoyFilter).Namespace(namespace).Get(
+//				context.TODO(),
+//				filtername,
+//				metav1.GetOptions{})
+//			if k8sapierrors.IsNotFound(err) == expectMissing {
+//				// filter state matches expected
+//				return true, nil
+//			}
+//
+//			//TODO If expectedMissing is true we could be going from F -> Nil or Nil -> F
+//			// Need to make sure the state stays as expected for some minimum time
+//		}
+//	}
+//}
```
```suggestion
```

sagikazarmark commented on test/e2e/e2e_helpers_test.go:
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
+		_, err := testEnv.Dynamic.Resource(gvr.EnvoyFilter).Namespace(namespace).Get(
+			context.TODO(),
+			filtername,
+			metav1.GetOptions{})
```
```suggestion
			metav1.GetOptions{},
		)
```

sagikazarmark commented on test/e2e/e2e_helpers_test.go:
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
 }
+
+func GetIstioObject(istio *istiov1beta1.Istio, namespace, name string) error {
+	return testEnv.Client.Get(context.TODO(), client.ObjectKey{
```
nit: coding style

sagikazarmark commented on test/e2e/e2e_helpers_test.go:
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
+			filtername,
+			metav1.GetOptions{})
+		if err != nil {
+			// log.Error(err, "Get filtername error")
```
???

sagikazarmark commented on test/e2e/e2e_suite_test.go:
```diff
@@ -17,10 +17,15 @@ limitations under the License.
 	"testing"
 	"time"
 
+	_ "encoding/json"
```
```suggestion
```

sagikazarmark commented on test/e2e/e2e_suite_test.go:
```diff
@@ -17,10 +17,15 @@ limitations under the License.
 import (
+	"bytes"
+	"encoding/json"
+	"github.com/kr/pretty"
```
```suggestion
```

sagikazarmark commented on test/e2e/e2e_suite_test.go:
```diff
@@ -17,10 +17,15 @@ limitations under the License.
 
+	_ "encoding/json"
 	"github.com/go-logr/logr"
+	_ "github.com/kr/pretty"
```
```suggestion
	"github.com/kr/pretty"
```

sagikazarmark commented on test/e2e/e2e_suite_test.go:
```diff
@@ -80,8 +85,33 @@ var _ = AfterSuite(func() {
+		var bmap, amap []byte
+		var err error
+		bmap, err = json.Marshal(map[string]ClusterResourceList{"ClusterStateBefore": clusterStateBefore})
+		if err != nil {
+			log.Error(err, "Problem with marshal")
+		} else {
+			err = json.Indent(&prettyJSON, bmap, "", "    ")
+			if err == nil {
+				log.Info("cluster resources before")
+				pretty.Print(string(prettyJSON.Bytes()))
+			}
+		}
```
```suggestion
		if err != nil {
			log.Error(err, "Problem with marshal")
			
			return
		}

		err = json.Indent(&prettyJSON, bmap, "", "    ")
		if err == nil {
			log.Info("cluster resources before")
			pretty.Print(string(prettyJSON.Bytes()))
		}
```

Similar changes are recommended below.

sagikazarmark commented on test/e2e/e2e_test.go:
```diff
@@ -17,7 +17,11 @@ limitations under the License.
 	"fmt"
+	"github.com/banzaicloud/istio-operator/pkg/util"
+	"strings"
+	//	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
```
```suggestion
```

sagikazarmark commented on test/e2e/e2e_test.go:
```diff
@@ -17,7 +17,11 @@ limitations under the License.
 import (
+	"context"
 	"fmt"
+	"github.com/banzaicloud/istio-operator/pkg/util"
```
Should move this down to the group below.

khernyo committed:
Fix resource name generation

The resource name is affected by the Istio.Spec.Global flag too: when
it's true (which it is by default), istio.Name is not appended to the
resource name. The proper name can be generated using
`istio.WithRevision()`.

khernyo committed:
Fix resource name

spongeman66 committed:
move common routines to helper and remove commented out code

spongeman66 commented:
a bad email was added to git, so unable to allow CLA.
This PR has been replaced by
https://github.com/banzaicloud/istio-operator/pull/663
